### PR TITLE
Feature/#7683

### PR DESCRIPTION
**Pull request by Moshyfawn**
Ref: #7683
In case you wanna support [the `focusout` event](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event)
As RHF doesn't officially support anything non-React, IDK if we wanna have tests with Preact.
Please, let me know, what you think.

_P.S. my 5 cents, RHF should support native Web event types as it's the Web API_

### CODE REVIEW

File: src/logic/createFormControl.ts [hunk 1]
```diff
@@ -655,7 +655,9 @@ export function createFormControl<
       const fieldValue = target.type
         ? getFieldValue(field._f)
         : getEventValue(event);
-      const isBlurEvent = event.type === EVENTS.BLUR;
+      const isBlurEvent = [EVENTS.BLUR, EVENTS.FOCUS_OUT].some(
+        (eventType) => event.type === eventType,
+      );
```
Comment (bluebill1049) [inline src/logic/createFormControl.ts:660]:
```suggestion
      const isBlurEvent = [EVENTS.BLUR, EVENTS.FOCUS_OUT].includes(event.type);
```

Save some bytes :)

**Review by bluebill1049 — COMMENTED** [review]
Thanks for the PR @Moshyfawn 🙏

File: src/logic/createFormControl.ts [hunk 2]
```diff
@@ -655,7 +655,7 @@ export function createFormControl<
       const fieldValue = target.type
         ? getFieldValue(field._f)
         : getEventValue(event);
-      const isBlurEvent = event.type === EVENTS.BLUR;
+      const isBlurEvent = [EVENTS.BLUR, EVENTS.FOCUS_OUT].includes(event.type);
```
Comment (bluebill1049) [inline src/logic/createFormControl.ts:658]:
```suggestion
      const isBlurEvent = event.type === EVENTS.BLUR || event.type === EVENTS.FOCUS_OUT;
```

Comment (bluebill1049) [reply src/logic/createFormControl.ts:658]: actually go with cheapest 👍

**Review by bluebill1049 — APPROVED** [review]
Merge and getting it published.

**Comment by Moshyfawn** [comment]
Fair enough!
Sry, it was night time for me :P

I wasn't sure about inducing any tests as, technically, in React there shouldn't be a `focusout` even, right?
```js
fireEvent.focusOut(screen.getByRole('textbox'));
```

**Comment by bluebill1049** [comment]
I think React still recognise focus event from the Dom.

**Comment by Moshyfawn** [comment]
I was referring [to this in React's codebase](React repo](https://github.com/facebook/react/blob/9198a5cec0936a21a5ba194a22fcbac03eba5d1d/packages/react-dom/src/shared/ReactDOMUnknownPropertyHook.js#L36)
```js
console.error(
  'React uses onFocus and onBlur instead of onFocusIn and onFocusOut. ' +
    'All React events are normalized to bubble, so onFocusIn and onFocusOut ' +
    'are not needed/supported by React.',
);
```

**Comment by bluebill1049** [comment]
I see, probably need to update that unit test then 👍

**Comment by Moshyfawn** [comment]
What do u wanna update?

**Comment by bluebill1049** [comment]
I think the test case just need to covert that we accept both events.

**Comment by Moshyfawn** [comment]
Idk, maybe it's good as a standalone test, since the `focusout` event is not a part of React. I don't really know what's the best strategy to test this case, tbh

**Comment by bluebill1049** [comment]
i will come back to it 

```tsx
const { onChange } = register()

onChange(pass a fake focus event)
```

**Pull request merged**